### ISSUE TITLE
Can lora also be implemented with stable diffusion?

### ISSUE THREAD

**Issue by staru09** [EricLBuehler/candle-lora#9]

**Comment by EricLBuehler** [comment]
Candle-lora can be applied to any model by adding the derive and attribute macros to each struct containing a linear, embedding, or conv layer. Therefore, it should be possible to apply candle-lora to any model.

**Comment by staru09** [comment]
https://github.com/huggingface/candle/tree/main/candle-transformers/src/models/stable_diffusion 
now there are a couple of files in this so in case I am planningng to port SD then do I have to port all these files or only some of the files?

**Comment by staru09** [comment]
ok let me try again then

**Comment by oovm** [comment]
Hello, have you successfully transplanted to stable diffusion?

I looked at some examples, but I don’t quite understand how AutoLoraConvert introduces LoRA weights.

**Comment by EricLBuehler** [comment]
@oovm, no, but you can do it by adding:
```rust
#[replace_layer_fields]
#[derive(AutoLoraConvert)]
```
to each model struct.

**Comment by staru09** [comment]
> Hello, have you successfully transplanted to stable diffusion?
> 
> I looked at some examples, but I don’t quite understand how AutoLoraConvert introduces LoRA weights.

I haven't already but it can be done by adding macro to every struct.

**Comment by EricLBuehler** [comment]
@donkey-donkey @staru09 yes exactly! Closing this as there hasn't been any activity for a bit.
